contracts/veBalFeeInjector.sol

```diff
@@ -75,16 +81,16 @@ contract veBalFeeInjector is ConfirmedOwner, Pausable {
    *
    */
   function performUpkeep(bytes calldata performData) external  onlyKeeperRegistry whenNotPaused {
-    bool upkeepNeeded;
+    uint256 timeCursor = feeDistributor.getTimeCursor();
+    require(LastRunTimeCurser < timeCursor, "Not ready");
+
+    int counter = 0;
     for(uint i=0; i<managedTokens.length; i++){
-      if (managedTokens[i].balanceOf(address(this)) > 0){
-        upkeepNeeded = true;
-      }
-      if (lastRunTimeCurser >= feeDistributor.getTimeCursor()) { //Not time yet
-        upkeepNeeded = false;
+      if (managedTokens[i].balanceOf(address(this)) > MinAmount * 10**(IERC20Metadata(address(managedTokens[i])).decimals())){
+        counter++;
       }
     }
-    require(upkeepNeeded, "Not ready");
+    require(counter == int256(managedTokens.length), "Not ready");
```

Comment:
```suggestion
    require(counter == managedTokens.length, "Not ready");
```
In this scenario `counter` cannot decrease, so making it an `uint` works. Applicable if you also change it here: https://github.com/BalancerMaxis/veBalFeeInjector/pull/9/files#r1238436885